src/cargo/core/compiler/build_context/mod.rs

```diff
@@ -69,11 +69,11 @@ impl<'a, 'cfg> BuildContext<'a, 'cfg> {
                 }
             })?;
             rustc.push_wrapper(RustcWrapper::new(tool));
-        } else if build_config.cargo_as_rustc_wrapper {
-            let mut wrapper = RustcWrapper::new(env::current_exe()?);
-            let prog = rustc.path.as_os_str().to_owned();
-            wrapper.env("RUSTC", prog);
-            rustc.push_wrapper(wrapper);
+            // } else if build_config.cargo_as_rustc_wrapper {
```

Comment:
it doesn't seem like we can use the cargo_as_rustc_wrapper stuff in the same way we do the clippy rustc wrapper.

If I understand this correctly, `cargo fix` wants go replace the `rustc` word in the cmdline and take a special env variable for `RUSTC` whereas other wrappers just go infront of it with `rustc` as their first argument.